nucypher/network/server.py

```diff
@@ -246,23 +246,23 @@ def set_policy(id_as_hex):
             except TimeExhausted:
                 # Alice didn't pay.  Return response with that weird status code.
                 this_node.suspicious_activities_witnessed['freeriders'].append((alice, f"No transaction matching {tx}."))
-                return Response(status=402)
+                return Response(f"No paid transaction matching {tx} for this node", status=402)
 
             this_node_has_been_arranged = this_node.checksum_address in arranged_addresses
             if not this_node_has_been_arranged:
                 this_node.suspicious_activities_witnessed['freeriders'].append((alice, f"The transaction {tx} does not list me as a Worker - it lists {arranged_addresses}."))
-                return Response(status=402)
+                return Response(f"No transaction arrangement matching {tx} for this node", status=402)
         else:
             _tx = NO_BLOCKCHAIN_CONNECTION
             kfrag_bytes = cleartext
         kfrag = KFrag.from_bytes(kfrag_bytes)
 
         if not kfrag.verify(signing_pubkey=alices_verifying_key):
-            raise InvalidSignature("{} is invalid".format(kfrag))
+            return Response(f"Signature on {kfrag} is invalid", status=403)
 
         with datastore.describe(PolicyArrangement, id_as_hex, writeable=True) as policy_arrangement:
             if not policy_arrangement.alice_verifying_key == alice.stamp.as_umbral_pubkey():
-                raise alice.SuspiciousActivity
+                return Response(f"Policy key does not match Alice stamp.", status=403)
```

Comment:
Tricky wording here. Perhaps something like:

```suggestion
                return Response("Policy arrangement's signing key does not match sender's", status=403)
```